### PR TITLE
empty check and yoda conditionals

### PR DESCRIPTION
**Pull request by willgladstone**

### CODE REVIEW

File: php/class-fm-bylines.php [hunk 1]
```diff
@@ -634,33 +634,34 @@ public function get_the_byline_meta( $field, $byline_id = null ) {
 				$byline_ids = $this->get_byline_ids();
 				$byline_id = reset( $byline_ids );
 			}
+			if ( ! empty( $byline_id ) ) {
```
Comment (damonsharp) [inline php/class-fm-bylines.php:638]: Just use else here? Bylines empty or not.

File: php/class-fm-bylines.php [hunk 2]
```diff
@@ -634,33 +634,34 @@ public function get_the_byline_meta( $field, $byline_id = null ) {
...
-
-			if ( $byline->post_type == $this->name ) {
-				$fields = $this->byline_meta_keys();
-				if ( ! empty( $fields[ $field ] ) ) {
-					$field_key = $fields[ $field ];
-				} else {
-					// This allows you to retrieve any meta key.
-					$field_key = $field;
-				}
+				if ( $byline->post_type == $this->name ) {
```
Comment (damonsharp) [inline php/class-fm-bylines.php]: Make sure $byline not null or $byline->post_type not empty before comparison.

File: php/class-fm-bylines.php [hunk 3]
```diff
@@ -634,33 +634,34 @@ public function get_the_byline_meta( $field, $byline_id = null ) {
...
 
-				$byline_meta = get_post_meta( $byline_id );
-				foreach ( $byline_meta as $key => $value ) {
-					if ( $field_key == $key ) {
-						return maybe_unserialize( $value[0] );
-					} elseif ( is_serialized( $value[0] ) ) {
-						$meta_values = maybe_unserialize( $value[0] );
-						if ( array_key_exists( $field_key, $meta_values ) ) {
-							return $meta_values[ $field_key ];
+					$byline_meta = get_post_meta( $byline_id );
```
Comment (damonsharp) [inline php/class-fm-bylines.php:657]: Check empty on $byline_meta before looping

**Comment by damonsharp** [comment]
Couple more checks, then 🎱

File: php/class-fm-bylines.php (hunk 1 above)
Comment (willgladstone) [reply php/class-fm-bylines.php:638]: It's possible this could still be entry. The above code simply tries to set an id if one is not passed.

**Pull request merged**